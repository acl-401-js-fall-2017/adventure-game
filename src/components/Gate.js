import React, { Component } from 'react';
import gatePic from '../pics/wonkaGate.jpg';
import PropTypes from 'prop-types';


export default class Gate extends Component {
  constructor() {
    super();
    this.state = {
      gateMessage: 'Welcome to the Chocolate factory',
    };
  }

  handleClick(value, inv){
    const { removeInventory, inventory } = this.props;
    this.setState({
      message: inventory.indexOf('golden-ticket') === -1 ? 'You need a ticket to get in' : this.state.gateMessage
    },() =>  {
      if(inventory.indexOf('golden-ticket') > -1) {
        removeInventory();
      }
    });
  }

  render() {
    
    return (
      <section className="challenge">
        <img src={gatePic} alt="" useMap="#gateMap" />
        <map name="gateMap" id="Map">
          <area data-value="correct" alt="grid" shape="poly" coords="479,147,477,566,636,567,622,156" 
            onClick={({ target }) => this.handleClick(target.dataset.value)} />
        </map>
        <p className="mission">
          Do you have a ticket to give Mr Wonka?
        </p>
        <p className="message">
          Message: {this.state.message}
        </p>
      </section>
    );
  }

}

Gate.PropTypes = {
  message: PropTypes.string
};
