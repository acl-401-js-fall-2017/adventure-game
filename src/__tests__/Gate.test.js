import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Gate from '../components/Gate';

describe('gate', () => {

  it('shows a Gate', () => {
    const wrapper = shallow(<Gate handleInventory={() => {}} inventory={[]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});