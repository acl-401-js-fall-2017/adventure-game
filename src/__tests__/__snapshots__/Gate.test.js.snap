// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gate shows a Gate 1`] = `
<section
  className="challenge"
>
  <img
    alt=""
    src="wonkaGate.jpg"
    useMap="#gateMap"
  />
  <map
    id="Map"
    name="gateMap"
  >
    <area
      alt="grid"
      coords="479,147,477,566,636,567,622,156"
      data-value="correct"
      onClick={[Function]}
      shape="poly"
    />
  </map>
  <p
    className="mission"
  >
    Do you have a ticket to give Mr Wonka?
  </p>
  <p
    className="message"
  >
    Message: 
  </p>
</section>
`;
